### ISSUE TITLE
Helicopter/VTOL and Tank AI modes.

### ISSUE THREAD

**Issue by RedReflection** [PapaJoesSoup/BDArmory#354]
- Feature Idea 
 - Enhancement 

A button in the hanger that can select which type of craft the AI will be piloting. 

Helicopter - thrust perpendicular to velocity. 

VTOL - some thrust vertical and some horizontal ( may or may not transition to full horizontal thrust ) Throttle Controlled Avionics can do a lot of the hard work, but the TCA pilot and the BDA pilot don't play well with each other.

Tank/Ship - Land or Sea based. Should be able to move to engage, hold a patrol route, or defend an area. 

In addition, AIs should be able to hold and defend an area, rather than relentlessly pursuing into danger/away from safety.

**Comment by MuricanJeb** [comment]
I don't really know about the VTOL. Does the AI know how to change modes for a VTOL engine? VTOL engines usually use a firespitter plugin to have their function, and BDA doesn't play that nice with firespitter props.

**Comment by jrodrigv** [comment]
OMG that is such an epic feature in terms of complexity . Each of them is
epic on its own:

Tank/Ship - Land or Sea based . *I would expect this one to be implemented
first. Maybe during 2018.*



2017-11-06 15:29 GMT+01:00 Murican_Jeb <notifications@github.com>:

**Comment by RedReflection** [comment]
Can't we manually tell it which mode is which? Say, defaults to vertical thrust, then switch mode for horizontal?

**Comment by RedReflection** [comment]
As far as land-based is concerned, can't we just strip out climbing and altitude based behaviour from the existing AI? Hell, as a starting point all a basic land/sea AI would have to do is just drive in a straight line, and turn every once in a while.

**Comment by MuricanJeb** [comment]
Yes, but then pitch would probably have to be configured as the "throttle." We should also have a way point system so the AI goes where we want to and stops following the way point route for combat, then returns to the way point route, and then we have to make the AI use roll (or yaw, depends on your settings for your vessel. However, this could be an option in the AI itself) so it can steer and then...

Lots of ideas, but can we really code it right now?

**Comment by PapaJoesSoup** [comment]
It would seem to be easy, but the ease of use the user experiences belies the complexities "under the hood".

This is not to say that it not doable, but I suspect that it will not be as simple as you might think. 

 There are a myriad of conditions to consider, and a refactor of the AI code to leverage code reuse.

It is definitely on our roadmap however, never fear.

**Comment by jrodrigv** [comment]
I agree with @PapaJoesSoup  AI development is always hard.

The easiest one would be the land/sea AI. However, the VTOL....we would need to relay on a mod like TCA to manage the thrust properly and then play with different values to

**Comment by jrodrigv** [comment]
Sorry for the close/reopen...

**Comment by RedReflection** [comment]
Oh, that's ok, I totally get that it is not a trivial task. Just wanted to keep it on the radar. :)

**Comment by RichardDastardly** [comment]
The current AI couldn't handle any of this, *at all*. While new pilot AI isn't a bad idea, not any time soon unless someone's started on it quietly already ( or at least got further than the "had some ideas" stage I'm at ).

**Comment by talker2** [comment]
> I ended up implementing a (rather basic) ship AI (the ones floating in water), including a few code tweaks to make the whole thing work with multiple types of AI modules.
> 
> Here's a few posts of it in action: https://www.youtube.com/watch?v=oiRJnbxMMXg https://forum.kerbalspaceprogram.com/index.php?/topic/166568-naval-battles-defeat-other-player-ships/&do=findComment&comment=3230691 https://forum.kerbalspaceprogram.com/index.php?/topic/166568-naval-battles-defeat-other-player-ships/&do=findComment&comment=3231184 https://www.youtube.com/watch?v=wO1-XCkU9VQ
> 
> Would this be useful, should I create a pull request?

 yes please.
